lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -321,14 +315,27 @@ def set_aspect(self, aspect, adjustable=None, anchor=None, share=False):
         --------
         mpl_toolkits.mplot3d.axes3d.Axes3D.set_box_aspect
         """
-        if aspect != 'auto':
+        if aspect not in ['auto', 'equal']:
             raise NotImplementedError(
                 "Axes3D currently only supports the aspect argument "
-                f"'auto'. You passed in {aspect!r}."
+                f"'auto' or 'equal'. You passed in {aspect!r}."
             )
         super().set_aspect(
             aspect, adjustable=adjustable, anchor=anchor, share=share)
 
+        if aspect == 'equal':
+            self.set_box_aspect((1., 1., 1.))
+            v_intervals = np.vstack((self.xaxis.get_view_interval(),
+                                     self.yaxis.get_view_interval(),
+                                     self.zaxis.get_view_interval())).T
+            mean = np.mean(v_intervals, axis=0)
+            delta = np.max(np.max(v_intervals, axis=0) -
+                           np.min(v_intervals, axis=0))
+
+            self.set_xlim3d(mean[0] - delta / 2., mean[0] + delta / 2.)
+            self.set_ylim3d(mean[1] - delta / 2., mean[1] + delta / 2.)
+            self.set_zlim3d(mean[2] - delta / 2., mean[2] + delta / 2.)
```

Comment:
Looks like these might change the limits set by the user?

Reply:
Yes, this must necessarily happen to change the aspect ratios. If the user sets limits afterwards, that will also break the equal aspect ratios